### DISCUSSION TITLE
Dose as SimpleQuantity

### DISCUSSION THREAD

**Discussion by rdbossjr** [FirelyTeam/firely-net-sdk#1695] [Q&A] [answered]
Hello, We are upgrading from a 1.* version of Hl7.Fhir.STU3. We are getting a compile error when trying to convert a MedicationRequest dose to a SimpleQuantity. It does not seem like SimpleQuantity exists and the Type of Dose is now DataType. Here is the code before:
```
            if (!(dosageInstruction.Dose is Quantity quantity))
            {
                quantity = dosageInstruction.Dose as SimpleQuantity;
            }
            var unit = quantity?.Unit;
```
Thanks in advance for any help given.
David

**Comment by marcovisserFurore** [comment, marked as answer]
Hi David,

`SimpleQuantity` is actually not a datatype but a profile on `Quantity`. This is also stated in the FHIR standard: https://www.hl7.org/fhir/STU3/datatypes.html#SimpleQuantity.

In Firely SDK version <= 1.9 this was not correctly implemented. For version 2.0 and higher this has been revised. So the datatype of `dosageInstruction.Dose` is `Quantity`  and has profile `SimpleQuantity`. Which means that the datatype has been constraint. In this case a comparator is not allowed on `Quantity`.

Profiles are not generated as Poco, that is why you won't find `SimpleQuantity` anymore in the code.
